### DISCUSSION TITLE
Is there a way to save a trained Splink model (much like saving a trained Keras neural network)?

### DISCUSSION THREAD

**Discussion by jkginfinite** [moj-analytical-services/splink#1282] [Q&A] [answered]
Hello is there a way to save a trained Splink model with the weights fully trained? I.e. we can reload it for later use and NOT feed it training data in the future, but instead just use the saved model for inference? 

I know there's the function save_settings_to_json() but that just looks like it saves the settings file in json format. (i.e. you'd need to load the settings and retrain it on new data when reusing it)

**Comment by RobinL** [comment, marked as answer]
The save_settings_to_json() function actually does save the trained model weights. Now that you mention it, I can see why the method name is confusing - it doesn't _just_ save the settings it also saves the trained weights.

 The following example shows how to load them back in:
https://moj-analytical-services.github.io/splink/demos/tutorials/05_Predicting_results.htm

I believe there may also have been a recent change where you can also pass the path to the settings JSON when you instantiate the linker

**Reply by RossKen** [reply]
on this @RobinL - would it make sense to rename this function to `save_model_to_json()`?

**Reply by RobinL** [reply]
Yeah, that sounds good to me

**Reply by RossKen** [reply]
Cool, I will put in a quick PR

**Reply by jkginfinite** [reply]
thanks Robin

**Reply by aeshirey** [reply]
For those that found this discussion via a web search, the above link is 404 and has changed to https://moj-analytical-services.github.io/splink/demos/tutorials/05_Predicting_results.html (nested under the 'tutorials').

**Reply by RobinL** [reply]
thanks, have update the orig link
